bftengine/include/bftengine/SimpleClient.hpp

```diff
@@ -72,7 +73,8 @@ class SeqNumberGeneratorForClientRequests {
  public:
   static SeqNumberGeneratorForClientRequests* createSeqNumberGeneratorForClientRequests();
 
-  virtual uint64_t generateUniqueSequenceNumberForRequest() = 0;
+  virtual uint64_t generateUniqueSequenceNumberForRequest(
+      std::chrono::time_point<std::chrono::system_clock> now = std::chrono::system_clock::now()) = 0;
```

Comment:
this is not very elegant... Let's preserve the original method and instead use method overloading:
```
virtual uint64_t generateUniqueSequenceNumberForRequest() = 0;
virtual uint64_t generateUniqueSequenceNumberForRequest(std::chrono::time_point<std::chrono::system_clock> now) = 0;
```

Then in `generateUniqueSequenceNumberForRequest()`, you get `std::chrono::system_clock::now()` and pass it to the overloaded `generateUniqueSequenceNumberForRequest(now)`

Reply:
In order to test the class behaviours, I need flexibility in
'std::chrono::time_point<std::chrono::system_clock> now' variable, move it back and forward to achieve behaviours that I won't be able to achieve, because this class depends on time.

Reply:
what I'm suggesting achieves the same result as what you had proposed - simply by using method overloading

Reply:
ok, I understand what you meant.